### ISSUE TITLE
Loads wrong image from cache

### ISSUE THREAD

**Issue by Leandros** [nostra13/Android-Universal-Image-Loader#258]
I load my images like this:

``` java
DisplayImageOptions displayOptions = new DisplayImageOptions.Builder()
                .cacheInMemory()
                .cacheOnDisc()
                .build();

String imageUriGooglePlus = "drawable://" + R.drawable.social_googleplus;
String imageUriFacebook = "drawable://" + R.drawable.social_facebook;
String imageUriTwitter = "drawable://" + R.drawable.social_twitter;
String imageUriRss = "drawable://" + R.drawable.social_rss;

ImageLoader imageLoader = ImageLoader.getInstance();
ImageSize targetSize = new ImageSize(imageSizePx, imageSizePx);
imageLoader.loadImage(imageUriGooglePlus, targetSize, displayOptions, new SimpleImageLoadingListener() {

            @Override
            public void onLoadingComplete(String imageUri, View view, Bitmap loadedImage) {
                mSocialGooglePlus.setImageBitmap(loadedImage);
            }
        });

[...] to be continued with the other bitmaps
```

Global Config:

``` java
DisplayImageOptions defaultOptions = new DisplayImageOptions.Builder()
                .cacheInMemory()
                .cacheOnDisc()
                .build();

// Create global configuration and initialize ImageLoader with this configuration
ImageLoaderConfiguration config = new ImageLoaderConfiguration.Builder(getApplicationContext())
                .defaultDisplayImageOptions(defaultOptions)
                .taskExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
                .taskExecutorForCachedImages(AsyncTask.THREAD_POOL_EXECUTOR)
                .build();

ImageLoader.getInstance().init(config);
```

**Comment by Shashank066** [comment]
I observed the same problem happening with me. There are very large images in drawable which are to be shown in gridview. It shows the correct images, but sometimes after scrolling very fastly wrong image gets used from disk cache. Now even when next time the app is launched the wrong image from cache will be shown. It gets corrected only when universal image loader disk cache is cleared. I have used universal image loader in many projects and it worked correctly. It happens only with this particular case of loading large images from drawable in gridview with diskcaching enabled.
